.github/workflows/connect-integration-tests.yml

```diff
@@ -0,0 +1,198 @@
+      - id: versions
+        working-directory: ./integration
+        # The `jq` command is "output compact, raw input, slurp, split on new lines, and remove the last element". This results in a JSON array of Connect versions (e.g., ["2025.01.0", "2024.12.0"]).
+        run: |
+          versions=$(make print-versions | jq -c -Rs 'split("\n") | .[:-1]')
+          echo "Versions: $versions"
+          echo "versions=$versions" >> "$GITHUB_OUTPUT"
+
+  # Run the Connect integration tests for each extension against each Connect version
+  connect-integration-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15 # Max time to run the integration tests
+    needs: setup-integration-test
+    strategy:
+      # Do not fail fast so all extensions and Connect versions are processed
+      fail-fast: false
+      matrix:
+        extension: ${{ fromJson(inputs.extensions) }}
+        connect_version: ${{ fromJson(needs.setup-integration-test.outputs.versions) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ./.github/actions/connect-integration-test
+        id: test
+        with:
+          extension-name: ${{ matrix.extension }}
+          connect-version: ${{ matrix.connect_version }}
+          connect-license: ${{ secrets.CONNECT_LICENSE }}
+
+      - uses: actions/upload-artifact@v4
+        if: |
+          always() && 
+          steps.test.outcome != 'cancelled' && 
+          steps.test.outcome != 'skipped'
+        with:
+          name: ${{ matrix.extension }}-${{ matrix.connect_version }}-test-report          
+          path: integration/reports/*.xml
+          retention-days: 7
+
+  # Analyses the test result files and publishes the results on the GitHub Actions job summary page
+  integration-test-report:
+    needs: connect-integration-test
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    # Create a matrix so each extension gets its own test report
+    strategy:
+      fail-fast: false
+      matrix:
+        extension: ${{ fromJson(inputs.extensions) }}
+    permissions:
+      checks: write
+      pull-requests: write
+    # Only run if tests weren't skipped or cancelled
+    if: |
+      always() && 
+      !contains(needs.connect-integration-test.result, 'skipped') &&
+      !contains(needs.connect-integration-test.result, 'cancelled')
+    steps:
+      - uses: actions/download-artifact@v4
+        id: download
+        with:
+          path: artifacts
+          pattern: "${{ matrix.extension }}-*-test-report" 
+
+      - uses: EnricoMi/publish-unit-test-result-action@v2
+        if: ${{ steps.download.outputs.download-path != '' }}
+        with:
+          check_name: "Integration test results - ${{ matrix.extension }}"
+          comment_mode: off
+          files: "artifacts/**/*.xml"
+          report_individual_runs: true
+
+  # Provides a matrix of extensions that passed all of the Connect integration tests
+  collect-results:
+    needs: [connect-integration-test, setup-integration-test]
```

Comment:
Since `connect-integration-test` needs `setup-integration-test` already I don't think this does.